.cbmc-batch/source/make_common_data_structures.c

```diff
@@ -21,6 +21,50 @@
 #include <stdint.h>
 #include <stdlib.h>
 
+bool is_valid_array_list(struct aws_array_list *list) {
+    size_t required_size;
+    return aws_mul_size_checked(list->length, list->item_size, &required_size) == AWS_OP_SUCCESS
+    && list->current_size >= required_size
+    && list->item_size != 0
+    && (list->alloc == NULL || list->alloc == can_fail_allocator());
+    /* TODO: once we can check allocated size of buffer, also assert that
+     * list->current_size <= __CPROVER_buffer_size(list->data);
```

Comment:
Hmm, what even is the property you'd like to express here? Wouldn't you have to start with "is a valid pointer?" Is maybe all you need:
```
((list->current_size == 0 && list->data == 0) || __CPROVER_r_ok(list->data, list->current_size))
```